.github/workflows/javadocs_scripts/parse_pom_auth_library.py

```diff
@@ -0,0 +1,46 @@
+# Parses sdk-platform-java/gapic-generator-parent-bom/pom.xml for auth library version
+
+import sys
+import xml.etree.ElementTree as ET
+import os.path
+
+ns = {
+    "maven": "http://maven.apache.org/POM/4.0.0"
+}
+
+# Get the input file from the command line argument
+input_file = [sys.argv[1]]
+
+for pom_file in input_file:
+  # Parse the XML file
+  tree = ET.parse(pom_file)
+  root = tree.getroot()
+
+  # Find the google-auth.version property
+  property_element = root.find("./maven:properties/maven:google.auth.version", ns)
+  google_auth_version = property_element.text
+
+# Save the auth library artifact and version to `/site/data/variables.yaml`
+with open(os.path.join('./site/data/', 'variables.yaml'), 'a') as f:
+    f.write(f'google-auth-library: "v{google_auth_version}"\n')
+
+# Save the auth library artifact and version to handwritten_libraries_javadocs_modules.txt
+output_file = ".github/workflows/javadocs_scripts/handwritten_libraries_javadocs_modules.txt"
+
+with open(output_file, 'a') as f:
+  # Update the line in-place
+  updated_line = f'googleapis/google-auth-library-java google-auth-library-java v{google_auth_version} google-auth-library\n'
+
+  # Read the contents of the output file
+  with open(output_file, 'r') as f:
+    lines = f.readlines()
+
+  # Find the line to update
+  for i, line in enumerate(lines):
+    if line.startswith("googleapis/google-auth-library-java"):
```

Comment:
Auth is unfortunately an exception in general because the only place (AFAIK) to find the version of the auth library used in the libraries-bom is in the sdk-platform-java/gapic-generator-java-pom-parent/pom.xml file. 

This script parses that file to get the version and then updates the `handwritten_libraries_javadocs_modules.txt` file in-place for that version so it knows which version to checkout for javadoc generation.